index.js

```diff
@@ -17,14 +22,53 @@ async function run() {
     const dockerLabels = core.getInput('docker-labels', { required: false });
     const command = core.getInput('command', { required: false });
 
-    // Parse the task definition
-    const taskDefPath = path.isAbsolute(taskDefinitionFile) ?
-      taskDefinitionFile :
-      path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
-    if (!fs.existsSync(taskDefPath)) {
-      throw new Error(`Task definition file does not exist: ${taskDefinitionFile}`);
+    //New inputs to fetch task definition 
+    const taskDefinitionArn = core.getInput('task-definition-arn', { required: false }) || undefined;
+    const taskDefinitionFamily = core.getInput('task-definition-family', { required: false }) || undefined;
+    const taskDefinitionRevision = Number(core.getInput('task-definition-revision', { required: false })) || null;
+
+    let taskDefPath;
+    let taskDefContents;
+    let describeTaskDefResponse;
+    let params;
+    
+    if (taskDefinitionFile) {
+      core.warning("Task definition file will be used.");
+      taskDefPath = path.isAbsolute(taskDefinitionFile) ?
+        taskDefinitionFile : 
+        path.join(process.env.GITHUB_WORKSPACE, taskDefinitionFile);
+      if (!fs.existsSync(taskDefPath)) {
+        throw new Error(`Task definition file does not exist: ${taskDefinitionFile}`);
+      }
+      taskDefContents = require(taskDefPath);
+    } else if (taskDefinitionArn || taskDefinitionFamily || taskDefinitionRevision) {
+      if (taskDefinitionArn) {
+        core.warning("The task definition arn will be used to fetch task definition");
+        params = {taskDefinition: taskDefinitionArn};
+      } else if (taskDefinitionFamily && taskDefinitionRevision) {
+        core.warning("The latest revision of the task definition family will be provided");
+        params = {taskDefinition: `${taskDefinitionFamily}:${taskDefinitionRevision}` };
+      } else if (taskDefinitionFamily) {
+        core.warning("The latest revision of the task definition family will be provided");
+        params = {taskDefinition: taskDefinitionFamily};
+      } else if (taskDefinitionRevision) {
+        core.setFailed("You can't fetch task definition with just revision: Either use task definition, arn or family");
+      } else {
+        throw new Error('Either task definition ARN, family, or family and revision must be provided');
+      }
+
+      try {
+        describeTaskDefResponse = await ecs.describeTaskDefinition(params);
+      } catch (error) {
+        core.setFailed("Failed to describe task definition in ECS: " + error.message);
+        core.debug("Task definition contents:");
+        core.debug(JSON.stringify(taskDefContents, undefined, 4));
```

Comment:
Since taskDefContents is only defined after the catch block the core.debug in lines 64 and 65 will not be helpful. The core.debug lines mentioned should be placed after the variable `taskDefContents` is initialized with a value. which happens in line 68 with `taskDefContents = describeTaskDefResponse.taskDefinition;`